frontend/src/app/main/data/workspace.cljs

```diff
@@ -1329,49 +1330,47 @@
           (calculate-paste-position [state mouse-pos in-viewport?]
             (let [page-objects  (wsh/lookup-page-objects state)
                   selected-objs (map #(get paste-objects %) selected)
-                  has-frame?    (d/seek #(= (:type %) :frame) selected-objs)
                   page-selected (wsh/lookup-selected state)
                   wrapper       (gsh/selection-rect selected-objs)
                   orig-pos      (gpt/point (:x1 wrapper) (:y1 wrapper))]
-              (cond
-                has-frame?
-                (let [index     (cph/get-position-on-parent page-objects uuid/zero)
-                      delta     (gpt/subtract mouse-pos orig-pos)]
-                  [uuid/zero uuid/zero delta index])
 
+              (cond
+                ;; Pasting inside a frame
                 (selected-frame? state)
                 (let [frame-id (first page-selected)
                       frame-object (get page-objects frame-id)
 
                       origin-frame-id (:frame-id (first selected-objs))
-                      origin-frame-object (get page-objects origin-frame-id)
-
-                      ;; - The pasted object position must be limited to container boundaries. If the pasted object doesn't fit we try to:
-                      ;;    - Align it to the limits on the x and y axis
-                      ;;    - Respect the distance of the object to the right and bottom in the original frame
-                      margin-x (if origin-frame-object
-                                 (- (:width origin-frame-object) (+ (:x wrapper) (:width wrapper)))
-                                 0)
-                      margin-x (min margin-x (- (:width frame-object) (:width wrapper)))
-
-                      margin-y  (if origin-frame-object
-                                  (- (:height origin-frame-object) (+ (:y wrapper) (:height wrapper)))
-                                  0)
-                      margin-y (min margin-y (- (:height frame-object) (:height wrapper)))
-
-                      ;; Pasted objects mustn't exceed the selected frame x limit
-                      paste-x (if (> (+ (:width wrapper) (:x1 wrapper)) (:width frame-object))
-                                (+ (- (:x frame-object) (:x orig-pos)) (- (:width frame-object) (:width wrapper) margin-x))
-                                (:x frame-object))
-
-                      ;; Pasted objects mustn't exceed the selected frame y limit
-                      paste-y (if (> (+ (:height wrapper) (:y1 wrapper)) (:height frame-object))
-                                (+ (- (:y frame-object) (:y orig-pos)) (- (:height frame-object) (:height wrapper) margin-y))
-                                (:y frame-object))
-
-                      delta    (gpt/point paste-x paste-y)]
-
-                  [frame-id frame-id delta])
+                      origin-frame-object (get page-objects origin-frame-id)]
+                  (cond
```

Comment:
I don't like much this nested `cond`. Usually it's better to try to move the variables up and have only one `cond`

If you think that's more confusing change this `cond` to `if` because it has only one condition.